README.md

```diff
@@ -70,12 +67,209 @@ RPC.call('test.getJoke', {}).then((result) => {
+## Constructor options
+URL, reconnectTimeout = 1000
+
+Parameter          | Type    | Required | Description
+-------------------|---------|----------|------------
+`URL`              | string  | Yes      | Absolute or relative URL
+`reconnectTimeout` | number  | No       | Timeout for reconnecting, defaults to `1000` ms
+
+```js
+// Url can be relative (schema for websocket would be detected automatically 
+// depending on page http/https schema)
+var RelativeUrlRPC = new WSRPC('/ws/', 5000);
+
+// Absolute websocket url example
+var UnsecureRPC = new WSRPC('ws://example.com/ws', 5000);
+
+// Secure absolute websocket url example
+var SecureRPC = new WSRPC('wss://example.com/ws', 5000);
+```
+
+## Methods
+
+#### WSRPC.connect()
+Establishes connection with the server.
+
+```js
+var RPC = new WSRPC(url);
+RPC.connect();
+```
+
+#### WSRPC.destroy()
+Closes socket.
+
+```js
+var RPC = new WSRPC(url);
+
+var deferred = RPC.onEvent('onconnect');
+deferred.resolve = function() {
+    RPC.destroy();
+};
+
+RPC.connect();
+```
+
+#### WSRPC.state()
+Get current socket state as a string.
+
+Possible values: `CONNECTING`, `OPEN`, `CLOSING`, `CLOSED`.
+
+```js
+var RPC = new WSRPC(url);
+console.log(RPC.state());  // Displays CLOSED
+```
+
+#### WSRPC.stateCode()
+Get current socket state code (integer). Possible values:
+
+Code | State
+-----|---------
+0    | CONNECTING
+1    | OPEN
+2    | CLOSING
+3    | CLOSED
+
+```js
+var RPC = new WSRPC(url);
+console.log(RPC.stateCode());  // Displays 3
+```
+
+#### WSRPC.addRoute(name, callback)
+Register route on the client with specified name, route added later replaces 
+route added earlier with the same name.
+
+Parameter | Type
+----------|------
+name      | string
+callback  | function
+
+Callback would be called with `object` type parameter `data` containing 
+parameters from server. Callback return value would be received by server. 
+
+```js
+var RPC = new WSRPC(url);
+RPC.addRoute('askUser', function(data) {
+    // Data is object, containing parameters from server.
+    // Would display question ask user to enter response and return response 
+    // to the server. 
+    return { response: prompt(data.question) };
+});
+```
+
+#### WSRPC.deleteRoute(name)
+Remove specified client route.
+
+Parameter | Type
+----------|------
+name      | string
+
+```js
+var RPC = new WSRPC(url);
+RPC.addRoute('askUser', function() { return {} });
+RPC.deleteRoute('askUser');
+```
+
+#### WSRPC.call(route, params)
+Call server function with specified parameters.
+
+Parameter | Type
+----------|------
+route     | string
+params    | object
+
+```js
+var RPC = new WSRPC(url);
+RPC.connect();
+RPC.call('serverRoute', {
+    param1: 'value1', 
+    param2: 'value2'
+}).then((result) => {
+    alert(result);
+}, (error) => {
+    alert(error.type + '("' + error.message + '")');
+});
+```
+
+#### WSRPC.addEventListener(event, callback)
+Add callback for event. Returns eventId, that can be used later to remove 
+event.
```

Comment:
Event handler will be registered permanently. See onEvent for register one time event